components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```
@@ -0,0 +1,586 @@
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(updatingAction.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(
+                updatingAction.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType()
+                           );
+    }
+
+    @Test(priority = 9)
+    public void testUpdateActionWithoutEndpointConfiguration() throws ActionMgtException {
+
+        Action updatingAction = buildMockUpdatingActionWithoutEndpointConfiguration();
+        Action result = daoImpl.updateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), updatingAction, action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(updatingAction.getName(), result.getName());
+        Assert.assertEquals(updatingAction.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(
+                action.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType()
+                           );
+    }
+
+    @Test(priority = 10)
+    public void testUpdateActionEndpointAuthProperties() {
+
+        try {
+            Action result = daoImpl.updateActionEndpointAuthProperties(action.getId(),
+                    buildMockBasicAuthType(true),
+                    TENANT_ID);
+            Assert.assertEquals(AuthType.AuthenticationType.BASIC, result.getEndpoint().getAuthentication().getType());
+            Assert.assertEquals(2, result.getEndpoint().getAuthentication().getProperties().size());
+        } catch (Exception e) {
+            Assert.fail();
+        }
```

You can throw the exception from the method and test will fail if it's an exception